dev/com.ibm.ws.microprofile.config.1.3/src/com/ibm/ws/microprofile/config13/sources/OSGiConfigUtils.java

```diff
@@ -60,23 +63,24 @@
      */
     public static String getApplicationName(BundleContext bundleContext) {
         String applicationName = null;
-        ComponentMetaData cmd = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData();
+        if (FrameworkState.isValid()) {
```

Comment:
This is a lot of `if (FrameworkState.isValid())`

at least it's in one class....

Reply:
yeah this helps cope if the system is shutting down around us